### PR TITLE
[sw/ottf] Reduce the size of the FreeRTOS heap.

### PR DESCRIPTION
**Pull request by timothytrippel**
The FreeRTOS heap is allocated as a static array. Currently, our flash C runtime (that we use for testing) zeros out the .bss section of memory before transferring control flow to`main()`, see #9163. This results in a performance hit for tests that do not make use of the FreeRTOS OS layer, i.e., non-concurrency tests. This commit reduces the size of the FreeRTOS stack to 0x800u bytes for the time being.

This partially addresses #8015.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

File: sw/device/lib/testing/test_framework/FreeRTOSConfig.h [hunk 1]
```diff
@@ -25,12 +25,9 @@
 
 // Memory
 #define configCHECK_FOR_STACK_OVERFLOW 0
-#define configMINIMAL_STACK_SIZE 512  // in words
+#define configMINIMAL_STACK_SIZE 256  // in words
+#define configTOTAL_HEAP_SIZE ((size_t)0x1500u)
```
Comment (cfrantz) [inline sw/device/lib/testing/test_framework/FreeRTOSConfig.h:29]: Per our offline discussion, please investigate the malloc failure hook so we can print a useful message in the event that a test exceeds the configured heap size.

Comment (timothytrippel) [reply sw/device/lib/testing/test_framework/FreeRTOSConfig.h]: Didn't realize I already had the malloc failure hook (mostly) implemented, just disabled (silly me!)

I have added an additional commit to enable this malloc failure hook (and print a more useful message). Additionally I have added a stack overflow hook that prints a useful message as well.

Lmk what you think!

Marking this as fixed.

**Comment by tjaychen** [comment]
i am curious, long term, do we see any utility in a hardware function that will initialize or zero-ize a certain area as directed by software? Ie, you could tell the hardware to put random data, 0's or 1's into a range with start address x and end address y. 

I suppose a lot of this depends on how performant the processor and fabric is.  But our setup overall is decidedly NOT performant.

**Comment by timothytrippel** [comment]
> i am curious, long term, do we see any utility in a hardware function that will initialize or zero-ize a certain area as directed by software? Ie, you could tell the hardware to put random data, 0's or 1's into a range with start address x and end address y.
> 
> I suppose a lot of this depends on how performant the processor and fabric is. But our setup overall is decidedly NOT performant.

Yes, I think so. See #9163.

File: sw/device/lib/testing/test_framework/freertos_hooks.c [hunk 2]
```diff
@@ -2,18 +2,31 @@
...
 
+/**
+ * This is called if configUSE_MALLOC_FAILED_HOOK is set to 1 in
+ * FreeRTOSConfig.h, and a call to pvPortMalloc() fails.
+ */
 void vApplicationMallocFailedHook(void) {
-  // TODO: communicate this event back to the host.
-  LOG_INFO("Malloc Failed.");
-  taskDISABLE_INTERRUPTS();
+  LOG_INFO("FreeRTOS malloc failed.\nIncrease heap size in FreeRTOSConfig.h");
```
Comment (mcy) [inline sw/device/lib/testing/test_framework/freertos_hooks.c:19]: LOG doesn't handle newlines great; I think you can just do away with them.

**Review by mcy — APPROVED** [review]

File: sw/device/lib/testing/test_framework/freertos_hooks.c (hunk 2 above)
Comment (timothytrippel) [reply sw/device/lib/testing/test_framework/freertos_hooks.c]: Fixed.

**Review by alphan — APPROVED** [review]

**Pull request merged**